Analytical/Classes/Provider/BranchProvider.swift

```diff
@@ -39,9 +39,9 @@ public class BranchProvider : BaseProvider<Branch>, AnalyticalProvider {
         }
     }
     
-    public func initSession() {
+    public func initSession(with deepLinkHandler: (([String: Any]?, Error?) -> Void)? = nil) {
         instance.initSession(launchOptions: launchOptions, andRegisterDeepLinkHandler: { params, error in
-            
+            deepLinkHandler?(params as? [String: Any], error)
```

Comment:
Why do we execute the handler in our own handler? Couldn't we just pass it to the method? And also add `deepLinkHandler` as the option on setup. :)